### PR TITLE
Update documentation for remote variables

### PR DESCRIPTION
**Pull request by DanilaFe**
This PR removes some limitations that were fixed in 2.2, and adjusts the spec section on remote variables to not say they're not implemented.

Reviewed by @e-kayrakli -- thanks!

### CODE REVIEW

File: doc/rst/language/spec/locales.rst [hunk 1]
```diff
@@ -276,13 +276,29 @@ created in the locale where the task is running. Variables can be
...
+   .. code-block:: chapel
+
+     proc computeInitialValue() {
+       writeln(here.id);
+       return 42;
+     }
+
+     on Locales(1) var x: int = computeInitialValue();
+
+   results in the output ``1`` because the ``writeln`` statement is
```
Comment (e-kayrakli) [inline doc/rst/language/spec/locales.rst:303]: "results in the output 1" reads a bit rough. Consider "outputs 1".

Can you also add a `writeln(x.locale)` after this?

File: doc/rst/technotes/remote.rst [hunk 2]
```diff
@@ -111,10 +104,10 @@ of a host array into it:
...
    on here.gpus[0] var GpuA = A[GpuPortion];
 
-If the target locale is a GPU locale, and the initialization expression
-is either a promoted expression or a loop expression, it is transformed into
-a GPU kernel as it would be in an ``on`` block. Thus, the following line of
-code generates a kernel launch:
+Since the initialization expression is executed on the target locale,
+if the array is initialized with a GPU-eligible expression, this expression
+is transformed into a GPU kernel as it would be in an ``on`` block.
+Thus, the following line of code generates a kernel launch:
```
Comment (e-kayrakli) [inline doc/rst/technotes/remote.rst:110]: I understand that this is not part of this PR, but would it make sense to add link(s) to the GPU technote and/or the blog series right before "Implementation Details" below?

**Review by e-kayrakli — APPROVED** [review]
Looks good! Please consider my comments inline, but you can merge at will.

File: doc/rst/technotes/remote.rst (hunk 2 above)
Comment (DanilaFe) [reply doc/rst/technotes/remote.rst:110]: The GPU tech note is linked just above, so I'm inclined to leave this as is.
